[GFPCOL-2437] Add Delivery endpoint to the developer guide (#4)
GetFeedback/embedded-listener-docs: developer-guide/index.md

We’ll be gathering feedback from our custom-built _Pepperoni-app_, hence we decide to name this channel the _Pepperoni channel_.

### Retrieving the JSON configuration

> We can now retrieve the JSON configuration of the form and channel we just created. In order to retrieve the configuration, we head to the *Channel setup* page where we have two options:

Reviewer: ```suggestion
It is now possible to retrieve the JSON configuration of the form and channel we just created. In order to do so, we head to the *Channel setup* page where two options are displayed:
```

Author: done

> 1. we copy the JSON directly from this page. This way suites best scenarios where internet connection is limited

Reviewer 2: - Capitalize start of sentence.
- Add `.` to end of line.
- Change `way` to `method`

Author: done

> 2. We copy the URL to retrieve the JSON programmatically. This method is best to automatically keep your form up to date with its latest published version.

Reviewer: ```suggestion
1. We can copy the JSON configuration directly from this page. This option is better suited for scenarios where internet connection is limited.
2. We can copy the URL to retrieve the JSON configuration programmatically. This method is best to automatically keep your form up to date with its latest published version.
```

Author: done

> Both options lead to the same value as the JSON we can copy is the same as the one we can retrieve from the URL. Also, the URL is predictable since it is always composed as:

Reviewer: ```suggestion
Regardless of the approach we choose, the resulting JSON configuration is exactly the same.
Protip: the resource location is predictable and it adheres to the following pattern:
```

Author: I think "Protip" sounds a bit too slang and could be misunderstood (see [Etymology](https://en.wiktionary.org/wiki/protip)). I'd remove it.

Reviewer: Haha, yeah! Protip is a bit too slangy

```
https://delivery.eu.getfbk.com/channels/<CHANNEL_ID>/<LOCALE>
```

where:

> - `CHANNEL_ID` is the id of the channel associated to the form

Reviewer: ```suggestion
- `CHANNEL_ID` is the id of the channel associated to the form.
```

Author: done

> - `LOCALE` is the language code specifying the localization of the form. This is either a specific locale or the values `main`, an alias for the main language. Because this alias will always exists for every configuration, it may be used as a fallback in case a specific localized configuration is not found.

Reviewer: ```suggestion
- `LOCALE` is the language code specifying the localization of the form. This is either a specific locale or the value `main`, alias for "main language". Because this alias will always exists for every configuration, it may be used as a fallback in case a specific localized configuration is not found.
```

Author: done

The payload **CAN** have the following properties:

* `userInput` _object_: as documented above
* `submitted` _bool_: as documented above

### Securing your data

At GetFeedback we value the privacy and security of your data, for this reason, we do not allow unauthenticated requests to our feedback collection services.

In order to submit feedback items via HTTP requests, you always need to provide an access token. The procedure to request an access token is described in the following sections.

> > **Note:** while we require authentication for submitting feedback items, we don't enforce it upon form configuration retrieval.

Reviewer 2: Don't think that `enforce` is the correct wording here since it's simply not implemented?

Author: Yep, changed with

> **Note:** while we enforce authentication for submitting feedback items, we don't require it upon form configuration retrieval.

  }'
```

## Caveats

#### Rate limiting

The current rate-limit for total number of **requests per second** **is 10** to submit feedback items. Any request exceeding the limit will be blocked by our services.

If you’re planning on exporting a large set of data that might exceed the limit, please reach out to customer support to discuss your use case.

#### Exception handling

It’s the implementor’s responsibility to ensure that the outcome of a feedback submission did not result in an exception. Please note that an exception may be raised on a later stage as each feedback item accepted by our collection endpoint will be further processed.

Any unhandled exception may result in potential data-loss, hence, re-submitting failed requests in case of temporary service unavailability is a must if you want to guarantee the completeness of your data.

#### Content-type of JSON configurations of forms

> The HTTP response returned by a form configuration retrieval request returns the header `Content-type: application/octet-stream`. The correct MIME type is `application/json` and the content SHOULD be interpreted correspondingly. We expect to fix this in future versions of the service.

Reviewer 3: do we :kekw: ?
